### PR TITLE
Add a simple function scalar IC

### LINKED ISSUE #6309: ScalarFunctionIC is needed
As title. To do this `ScalarInitialCondition` needs to be derived from `FunctionInterface`. Do you want me create this IC in MOOSE or you prefer it is on my app?

### PR DESCRIPTION
**Pull request by YaqiWang**
Close #6309.

### CODE REVIEW

File: test/tests/ics/function_scalar_ic/function_scalar_ic.i [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    family = SCALAR
+    order = FIRST
+  [../]
+[]
+
+[Functions]
+ [./f]
+   type = ParsedFunction
+   value = cos(t)
+ [../]
```
Comment (andrsd) [inline test/tests/ics/function_scalar_ic/function_scalar_ic.i]: Indent by 2 spaces

File: framework/src/ics/FunctionScalarIC.C [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    mooseError("number of functions must be equal to the scalar variable order");
+
+  for (unsigned int i=0; i<funcs.size(); ++i)
+    _func.push_back(&getFunctionByName(funcs[i]));
+}
+
+Real
+FunctionScalarIC::value()
+{
+  return _func[_i]->value(_t, Point(0,0,0));
```
Comment (andrsd) [inline framework/src/ics/FunctionScalarIC.C]: Add spaces after commas.

File: framework/src/ics/FunctionScalarIC.C [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+FunctionScalarIC::FunctionScalarIC(const InputParameters & parameters) :
+    ScalarInitialCondition(parameters),
+    _ncomp(_var.order())
+{
+  std::vector<FunctionName> funcs = getParam<std::vector<FunctionName> >("function");
+  if (funcs.size() != _ncomp)
+    mooseError("number of functions must be equal to the scalar variable order");
+
+  for (unsigned int i=0; i<funcs.size(); ++i)
```
Comment (andrsd) [inline framework/src/ics/FunctionScalarIC.C]: Pad the operators with spaces.

File: test/tests/ics/function_scalar_ic/function_scalar_ic.i [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+[Executioner]
+ type = Transient
+ start_time = 0
+ end_time = 1
+ dt = 0.01
+ scheme = bdf2
+ solve_type = 'PJFNK'
+ timestep_tolerance = 1e-12
+[]
```
Comment (andrsd) [inline test/tests/ics/function_scalar_ic/function_scalar_ic.i]: Indent by 2 spaces

File: test/tests/ics/function_scalar_ic/function_scalar_ic.i [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+[Outputs]
+ csv = true
```
Comment (andrsd) [inline test/tests/ics/function_scalar_ic/function_scalar_ic.i]: Indent by 2 spaces

**Comment by YaqiWang** [comment]
@andrsd Your comments are addressed. Thanks!

**Pull request merged**